src/components/progress-indicators/WalletSyncCircle.tsx

```diff
@@ -26,29 +26,26 @@ export const WalletSyncCircle = (props: Props) => {
   const { size = theme.rem(2), wallet } = props
   // Animation shared state
   const syncRatio = useSharedValue(wallet.syncRatio < RESYNC_THRESHOLD ? MIN_RATIO : wallet.syncRatio)
-  const isDone = useSharedValue(false)
 
   // Subscribe to the sync ratio:
-  React.useEffect(
-    () =>
-      wallet.watch('syncRatio', (ratio: number) => {
-        // If already done but needs to resync reset the flags and animations
-        if (isDone.value && ratio < RESYNC_THRESHOLD) {
-          isDone.value = false
-          syncRatio.value = RESYNC_THRESHOLD
-        }
-        // If the wallet hasn't fully synced show progress animation
-        if (!isDone.value) {
-          // Sync is done
-          if (ratio > DONE_THRESHOLD) isDone.value = true
+  React.useEffect(() => {
+    const handler = (ratio: number) => {
+      // If already done but needs to resync reset the flags and animations
+      if (syncRatio.value > DONE_THRESHOLD && ratio < RESYNC_THRESHOLD) {
+        syncRatio.value = RESYNC_THRESHOLD
+        return
+      }
 
-          if (ratio < RESYNC_THRESHOLD) syncRatio.value = withTiming(RESYNC_THRESHOLD, { duration: 1000 })
-          else if (ratio > MAX_RATIO && ratio < DONE_THRESHOLD) syncRatio.value = withTiming(MAX_RATIO, { duration: 1000 })
-          else syncRatio.value = withTiming(ratio, { duration: 1000 })
-        }
-      }),
-    [isDone, syncRatio, wallet]
-  )
+      if (ratio < RESYNC_THRESHOLD) syncRatio.value = withTiming(RESYNC_THRESHOLD, { duration: 1000 })
+      else if (ratio > MAX_RATIO && ratio < DONE_THRESHOLD) syncRatio.value = withTiming(MAX_RATIO, { duration: 1000 })
+      else syncRatio.value = withTiming(ratio, { duration: 1000 })
+    }
```

Comment:
You have the `ratio < RESYNC_RATIO` logic twice, which makes this tricky to read. I would refactor this entire block as follows:

```ts
if (ratio < RESYNC_THRESHOLD) {
  // Do not animate backwards if a resync happens after the sync is done:
  if (syncRatio.value > DONE_THRESHOLD) {
    syncRatio.value = RESYNC_THRESHOLD
  } else {
    syncRatio.value = withTiming(RESYNC_THRESHOLD, { duration: 1000 })
  }
} else if (ratio > MAX_RATIO && ratio < DONE_THRESHOLD) {
  syncRatio.value = withTiming(MAX_RATIO, { duration: 1000 })
} else {
  syncRatio.value = withTiming(ratio, { duration: 1000 })
}
```

This consolidates the two cases involving RESYNC_THRESHOLD, and uses `else` for control flow instead of the early return. If you want to use returns for control flow, that would work too, but it should be consistent for all the cases.